src/tensorwaves/optimizer/callbacks.py

```diff
@@ -2,121 +2,85 @@
 
 from abc import ABC, abstractmethod
 from datetime import datetime
-from typing import IO, Iterable, List, Optional
+from typing import IO, Any, Dict, Iterable, List, Optional
 
 import pandas as pd
 import tensorflow as tf
 import yaml
-from tqdm import tqdm
 
-from tensorwaves.interfaces import Estimator
+
+class Loadable(ABC):
+    @staticmethod
+    @abstractmethod
+    def load_latest_parameters(filename: str) -> dict:
```

Comment:
why not make it
```python3
@abstractmethod
￼def latest_parameters() -> dict:
```

Reply:
Because it actually loads the latest parameters. The stream that's used by `on_iteration_end` is write mode and is also closed after `on_function_call_end` is called. Note also that `CSVSummary` (and eventually `TFSummary`) have a different implementation that does not use Python `open` (internally a stream).

Reply:
why cant the loading be done in a separate function that is not part of the "interface"?

Reply:
Ok, so the summary is that the `Loadable` interface is not crucial, as its not connecting modules together, but "only" to the user (UI). Hence there are no avalanches propagating through the code if this will be changed later on. For more flexibility (e.g: writing to some web based API), the current interface would have to be abolished. The tricky part when having a interface like `def latest_parameters() -> dict:` is that both the input and output are connected to the state of the class. This can be tricky when loading a file in read mode, but writing to it later on.